assignments/pull_requests.go

```diff
@@ -0,0 +1,98 @@
+package assignments
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/database"
+	"github.com/autograde/quickfeed/scm"
+	"github.com/google/go-github/v35/github"
+	"go.uber.org/zap"
+)
+
+var ErrInvalidBody = errors.New("invalid pull request body")
+
+// getLinkedIssue returns the issue number from a pull requests body.
+// E.g. 30, from the body "Fixes #30".
+func getLinkedIssue(body string) (uint64, error) {
+	if count := strings.Count(body, "#"); count != 1 {
+		return 0, ErrInvalidBody
+	}
+	_, numString, _ := strings.Cut(body, "#")
+	issueNumber, err := strconv.Atoi(numString)
+	if err != nil {
+		return 0, ErrInvalidBody
```

Comment:
Could probably either return the `err` directly or wrap `err` in a `fmt.Errorf("more context...: %w", err)`